src/carriers/bayer_carrier/BayerCarrier.h

```diff
@@ -65,20 +65,22 @@ class yarp::os::BayerCarrier : public yarp::os::ModifyingCarrier,
     // ModifyingCarrier methods
 
 
-    BayerCarrier() {
-        need_reset = true;
-        have_result = false;
-        image_data_len = 0;
-        half = false;
-        warned = false;
-        goff = 0;
-        roff = 1;
-        bayer_method = -1;
-        bayer_method_set = false;
-        dcformat = -1;
-        local = yarp::os::ConnectionReader::createConnectionReader(*this);
-        happy = (local!=0);
-    }
+    BayerCarrier() :
+        local(yarp::os::ConnectionReader::createConnectionReader(*this)),
+        parent(YARP_NULLPTR),
+        image_data_len(0),
```

Comment:
same here: `image_data_len` is declared before `local`

Reply:
Done, https://github.com/robotology/yarp/pull/1412/commits/33fbf8e89f881d979d867e75382cec8415401b77